src/ompl/base/samplers/DeterministicStateSampler.h

```diff
@@ -0,0 +1,172 @@
+ *  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ *  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ *********************************************************************/
+
+/* Author: Leonard Bruns */
+
+#ifndef OMPL_BASE_DETERMINISTIC_STATE_SAMPLER_
+#define OMPL_BASE_DETERMINISTIC_STATE_SAMPLER_
+
+#include "ompl/base/StateSpace.h"
+#include "ompl/base/StateSampler.h"
+#include "ompl/base/samplers/deterministic/DeterministicSequence.h"
+
+#include <memory>
+
+namespace ompl
+{
+    namespace base
+    {
+        /**
+       @anchor DeterministicStateSampler
+       @par Short description
+       \ref DeterministicStateSampler Implementation of a deterministic state sampler. The
+       implementation allows to load and draw samples from a precomputed sequence or from the Halton Sequence,
+       @par External documentation
+       Dispertio: Optimal Sampling For Safe Deterministic Motion Planning
+       L Palmieri, L Bruns, M Meurer, KO Arras - IEEE Robotics and Automation Letters, 2019
+       DOI: [10.1109/LRA.2019.2958525](https://ieeexplore.ieee.org/abstract/document/8928532)<br>
+       [[PDF]](https://ieeexplore.ieee.org/stamp/stamp.jsp?tp=&arnumber=8928532)
+    */
+
+        /** \brief An abstract class for the concept of using deterministic sampling sequences
+        to decrease the dispersion of the samples. */
+        class DeterministicStateSampler : public StateSampler
+        {
+        public:
+            enum DeterministicSamplerType
+            {
+                HALTON
+            };
+
+            /** \brief Constructor, which creates the sequence internally based on the specified sequence type.
+            Uses the default constructor for the sequence.*/
+            DeterministicStateSampler(const StateSpace *space,
+                                      DeterministicSamplerType type = DeterministicSamplerType::HALTON);
+            /** \brief Constructor that takes a pointer to a DeterministicSequence and uses that object instead
+            of its own. This can be used to apply non default options to the deterministic sequence.*/
+            DeterministicStateSampler(const StateSpace *space, std::shared_ptr<DeterministicSequence> sequence_ptr);
+
+            void sampleUniform(State *state) override = 0;
+            void sampleUniformNear(State *state, const State *near, double distance) override = 0;
```

Comment:
Provide a default implementation for this method and the next one that calls OMPL_ERROR to say this is unsupported. This will eliminate these methods in the derived classes.

Reply:
Implemented